js_modules/dagster-ui/packages/ui-core/src/hooks/useStateWithStorage.tsx

```diff
@@ -51,8 +51,11 @@ export function useStateWithStorage<T>(key: string, validate: (json: any) => T)
 
   const setState = React.useCallback(
     (input: React.SetStateAction<T>) => {
-      const next =
+      let next: any =
         input instanceof Function ? input(validateRef.current(getJSONForKey(key))) : input;
+      if (next instanceof Set) {
+        next = Array.from(next);
```

Comment:
Wow TIL `JSON.stringify(new Set(['A', 'B']))` does not work.  Good catch!  I wish it wasn't so hard to adopt Set... I feel like it's the "correct" data structure in many places, but things like this make me dislike using it anywhere the volume of items is low :(